src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/CsvFormatResponseIT.java

```diff
@@ -142,6 +156,460 @@ public void fieldOrderWithScriptFields() throws IOException {
         verifyFieldOrder(expectedFields, query);
     }
 
+    //region Tests migrated from CSVResultsExtractorTests
+
+    @Test
+    public void simpleSearchResultNotNestedNotFlatNoAggs() throws Exception {
+        String query = String.format(Locale.ROOT, "select dog_name,age from %s/dog order by age", TEST_INDEX_DOG);
+        final CSVResult csvResult = executeCsvRequest(query, false);
+
+        List<String> headers = csvResult.getHeaders();
+        Assert.assertEquals(2, headers.size());
+        Assert.assertTrue("name should be on headers", headers.contains("dog_name"));
+        Assert.assertTrue("age should be on headers", headers.contains("age"));
+
+        List<String> lines = csvResult.getLines();
+        Assert.assertEquals(2, lines.size());
+        Assert.assertTrue("rex,2".equals(lines.get(0)) || "2,rex".equals(lines.get(0)));
+        Assert.assertTrue("snoopy,4".equals(lines.get(1)) || "4,snoopy".equals(lines.get(1)));
+    }
+
+    @Test
+    public void simpleSearchResultWithNestedNotFlatNoAggs() throws Exception {
+        String query = String.format(Locale.ROOT, "select name,house from %s/gotCharacters",
+                TEST_INDEX_GAME_OF_THRONES);
+        CSVResult csvResult = executeCsvRequest(query, false);
+
+        List<String> headers = csvResult.getHeaders();
+        Assert.assertEquals(2, headers.size());
+        Assert.assertTrue("name should be on headers", headers.contains("name"));
+        Assert.assertTrue("house should be on headers", headers.contains("house"));
+
+        List<String> lines = csvResult.getLines();
+        Assert.assertEquals(7, lines.size());
+        Assert.assertTrue(lines.contains("\"{firstname=Daenerys, lastname=Targaryen, ofHerName=1}\",Targaryen") ||
```

Comment:
Are we trying to permute them here? If so, I think we better use Guava (`Collections2.permutations()`) to generate permutation. And adding a helper method would be useful. For example,

```
private boolean containsAny(String lines, Collection<String> strs) {} // You can come up with better names than strs :)
```

Reply:
The method you mentioned is marked as beta and is unstable, we should not use it. I wrote my own getPermutations, will update the tests with that one.